Source/WebKit/UIProcess/Cocoa/UserMediaCaptureManagerProxy.cpp

```diff
@@ -647,12 +647,15 @@ void UserMediaCaptureManagerProxy::takePhoto(RealtimeMediaSourceIdentifier sourc
 
 void UserMediaCaptureManagerProxy::getPhotoCapabilities(RealtimeMediaSourceIdentifier sourceID, GetPhotoCapabilitiesCallback&& handler)
 {
-    if (auto* proxy = m_proxies.get(sourceID)) {
-        proxy->getPhotoCapabilities(WTFMove(handler));
+    auto* proxy = m_proxies.get(sourceID);
+    if (!proxy) {
+        handler(Unexpected<String>("Device not available"_s));
         return;
     }
 
-    handler(PhotoCapabilitiesOrError("Device not available"_s));
+    proxy->getPhotoCapabilities()->whenSettled(RunLoop::main(), [handler = WTFMove(handler)] (auto&& result) mutable {
```

Comment:
` proxy->getPhotoCapabilities()->whenSettled(RunLoop::main(), WTFMove(handler));`

should work I think

Reply:
Of course!